src/dstruct/factor_graph/factor_graph.cpp

```diff
@@ -82,25 +82,36 @@ long dd::FactorGraph::get_multinomial_weight_id(
   /**
    * The weight ids are aligned in a continuous region according
    * to the numerical order of variable values.
-   * Say for two variables v1, v2, v3, with cardinality d. The numerical value
-   * is
-   * v1 * d^2 + v2 * d + v3.
+   * For example, for three variables v1, v2, v3, with cardinality d1, d2, and
+   * d3.
+   * The weight index is
+   * v1 * d2 * d3 + v2 * d3 + (index of v3 in domain vector).
```

Comment:
Something like this will be easier to parse visually:
```
(...(((0 * d1 + v1) * d2 + v2) * d3 + v3)...) * dk + vk
```
which also mirrors the code exactly.